### ISSUE TITLE
RTO C# MarketPriceRdpGwClientCredAuthExample shows "KeyNotFoundException: The given key 'Session1' was not present in the dictionary."

### ISSUE THREAD

**Issue by wasin-waeosri** [thomsonreuters/websocket-api#21]
Hi team,

When I run the RTO C# MarketPriceRdpGwClientCredAuthExample using the Service Discovery feature, the example application throws the "System.Collections.Generic.KeyNotFoundException: The given key 'Session1' was not present in the dictionary." exception  message and crashes:

Error Messages: 
```
dotnet MarketPriceRdpGwClientCredAuthExample.dll --clientid $RTO_CLIENTID_V2_1 --clientsecret $RTO_CLIENTSECRET_1 
9/12/22 10:52:38 AM Sending authentication request to https://api.refinitiv.com/auth/oauth2/v2/token
9/12/22 10:52:40 AM RECEIVED:
{
  "expires_in": 7199,
  "token_type": "Bearer",
  "access_token": "<access_token>"
}

9/12/22 10:52:41 AM Sending service discovery request to https://api.refinitiv.com/streaming/pricing/v1/?transport=websocket

System.Collections.Generic.KeyNotFoundException: The given key 'Session1' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at MarketPriceRdpGwServiceDiscoveryExample.MarketPriceRdpGwServiceDiscoveryExample.Run() in /opt/refinitiv/websocket-api/Applications/Examples/RDP/CSharp/MarketPriceRdpGwClientCredAuthExample/MarketPriceRdpGwClientCredAuthExample.cs:line 644
```
However, when I specify the ```--hostname``` argument, it works fine.

Example:
```
dotnet MarketPriceRdpGwClientCredAuthExample.dll --clientid $RTO_CLIENTID_V2_1 --clientsecret $RTO_CLIENTSECRET_1 --hostname ap-northeast-1-aws-1-sm.optimized-pricing-api.refinitiv.net
```

According to my observation, if I do not set the ```--hostname``` argument (use the Service Discovery), the following code statement is not executed.

```
 foreach (var host in _hosts)
                {
                    var webSocketSession = new WebSocketSession("Session" + (_webSocketSessions.Count + 1), host.Item1, host.Item2);
                    _webSocketSessions.Add(webSocketSession.Name, webSocketSession);
...
```

Best regards,

Wasin W.

**Comment by L-Karchevska** [comment]
@wasin-waeosri  Thank you bringing this issue to our attention, we will investigate and fix it in the future releases.

**Comment by vlevendel** [comment]
Addressed with tag WSA-1.1.10
